packages/xo-server/src/xapi/index.js

```diff
@@ -734,15 +734,25 @@ export default class Xapi extends XapiBase {
       const { SR } = vdi
       let childrenMap = cache[SR]
       if (childrenMap === undefined) {
-        childrenMap = cache[SR] = groupBy(
-          vdi.$SR.$VDIs,
-          _ => _.sm_config['vhd-parent']
-        )
+        try {
+          childrenMap = cache[SR] = groupBy(
+            vdi.$SR.$VDIs,
+            _ => _.sm_config['vhd-parent']
+          )
+        } catch (error) {
+          log.warn('_assertHealthyVdiChain', { error })
+        }
       }
 
       // an unmanaged VDI should not have exactly one child: they
       // should coalesce
-      const children = childrenMap[vdi.uuid]
+      const children =
+        childrenMap !== undefined
+          ? childrenMap[vdi.uuid]
+          : vdi.$SR.$VDIs.filter(
+              _ => _ !== undefined && _.sm_config['vhd-parent'] === vdi.uuid
```

Comment:
it would be better if you describe the `_` symbol, e.g. `srVdi`

Reply:
IMHO, in this case, the meaning of the symbol is clear.
We filter `$VDIs` collection, so, an element in the collection we'll only be a VDI which is contained on a SR.
This syntax is also used [here](https://github.com/vatesfr/xen-orchestra/pull/4417/files#diff-0d7c8183f21c8290ffbeb18bca5c0f88L739).

Reply:
Ok that seems good to me.

Reply:
@HamadaBrest When one of your comment is *resolved*, please mark it as such.